plugins/woocommerce/includes/class-wc-download-handler.php

```diff
@@ -580,13 +583,54 @@ private static function check_server_config() {
 	 * Can prevent errors, for example: transfer closed with 3 bytes remaining to read.
 	 */
 	private static function clean_buffers() {
-		if ( ob_get_level() ) {
-			$levels = ob_get_level();
-			for ( $i = 0; $i < $levels; $i++ ) {
-				@ob_end_clean(); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged
+		$levels = ob_get_level();
+		for ( $i = 0; $i < $levels; $i++ ) {
+			$ob_status = ob_get_status();
+
+			// can be not set in some installations
+			if ( ! isset( $ob_status['flags'] ) ) {
+				return;
 			}
-		} else {
-			@ob_end_clean(); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged
+
+			if ( ( $ob_status['flags'] & PHP_OUTPUT_HANDLER_REMOVABLE ) === PHP_OUTPUT_HANDLER_REMOVABLE ) {
+				ob_end_clean();
+				continue;
+			}
+
+			if ( ( $ob_status['flags'] & PHP_OUTPUT_HANDLER_CLEANABLE ) === PHP_OUTPUT_HANDLER_CLEANABLE ) {
```

Comment:
Might this be a little more readable? 

```suggestion
			if ( $ob_status['flags'] & PHP_OUTPUT_HANDLER_CLEANABLE ) {
```